muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -1045,4 +1086,175 @@ else if (value instanceof Double) {
     return date;
   }
 
+  @POST
+  @Path("/student/{ID}/workspace/{WORKSPACEID}/signup")
+  @RESTPermit (handling = Handling.INLINE)
+  public Response createWorkspaceSignupByStaff(@PathParam("ID") String userIdentifier, @PathParam("WORKSPACEID") Long workspaceEntityId,
+      fi.otavanopisto.muikku.plugins.workspace.rest.model.WorkspaceUserSignup entity) {
+
+    if (!sessionController.isLoggedIn()) { 
+      return Response.status(Status.UNAUTHORIZED).build();
+    }
+    
+    SchoolDataIdentifier studentIdentifier = SchoolDataIdentifier.fromId(userIdentifier);
+
+    UserEntity studentEntity = userEntityController.findUserEntityByDataSourceAndIdentifier(studentIdentifier.getDataSource(), studentIdentifier.getIdentifier());
```

Comment:
Could be just `userEntityController.findUserEntityByUserIdentifier(studentIdentifier)`